core/web/log_controller.go

```diff
@@ -50,6 +51,15 @@ func (cc *LogController) Patch(c *gin.Context) {
 		}
 	}
 
+	if request.Filter != "" {
+		cc.App.GetStore().Config.Set("LOG_FILTER", request.Filter)
+		err := cc.App.GetStore().SetConfigStrValue("LogFilter", request.Filter)
```

Comment:
nice, though i think the issue that remains after these, is about actually replacing the loggers globally in the app. Otherwise the dynamic change of logging will not have any immediate effect.
I suppose the first step would be to re-create the global variable `logger.Default` (probably via some improvement to `logger.SetLogger(logger.CreateProductionLogger(....))`

But still, this will not be per-package logging, as per-package/service approach only works if there was  a specific name set for a logger. So after updating `logger.Default`, somehow all (or just a few for now) services need to be notified to take the new `logger.Default` and re-create a custom named loggers from it

Reply:
ah i see, I'm setting the global `logger.Default` with `	logger.Default = cc.App.GetStore().Config.CreateProductionLogger()` which should configure the log levels with the names provided in the filter string. 

So if I understand correctly, we need a standard naming for the logging pkgs/services and have them re-initialized in the pkg context as well when this is updated?

Reply:
`logger.Default = cc.App.GetStore().Config.CreateProductionLogger()` - hmm i can't see this in this diff?

but yeah you need to re-create even the default logger after each log level change (per-package or the global one) 

```
So if I understand correctly, we need a standard naming for the logging pkgs/services and have them re-initialized in the pkg context as well when this is updated?
```
yes, with probably something like `logger := logger.Default.Named("flux_monitor")` in each service

Reply:
This pr is based off https://github.com/smartcontractkit/chainlink/pull/4065 which has the change for re-setting the logger from the endpoint: 

https://github.com/smartcontractkit/chainlink/pull/4065/files#diff-679da40bfd15572c69367c79980c1840cb020b5961776a9ab5db2a4628abd12eR64

```
not sure what's the best way to notify each service - either call a method on them or (more likely) send a message to them using the per-service mailbox, so avoid any threading issues
```

Some kind of channel makes sense, does each service have one available? Mailbox looks created on each service separately, know if theres a global way to call? Looking into it

Reply:
ah ok - commented there now about `logger.SetLogger`

Reply:
`Mailbox looks created on each service separately, know if theres a global way to call? Looking into it` might need  a new mailbox on each service. i think generally they have a mailbox per message type

Reply:
Perhaps it would make more sense to instantiate a new logger for each service and pass in that pointer at struct creation time. This has the added benefit of allowing each service to set global structured logging for itself as well.

Then we can collect the pointers for all the loggers and peg them onto the application or something where they can be accessed globally.